### DISCUSSION TITLE
Stacked merge requests on Gitlab

### DISCUSSION THREAD

**Discussion by JaniM** [arxanas/git-branchless#1248] [Show and tell]
I wanted to automate managing stacked PRs on Gitlab. After some manual-reading and tinkering, it turned out surprisingly simple.

First, @pokey's [auto-branch script](https://github.com/pokey/dotfiles/blob/f7e3c8c632cd898406c8bc56fc4eb859d416e58e/bin/auto-branch) produces branches for each commit in the current stack.
Then, my [push-prs-gitlab script](https://github.com/JaniM/dotfiles-mac/blob/e6216e0af76337b35c918af6c4f8d02a9a97558b/scripts/push-prs-gitlab) pushes each branch in the stack to Gitlab and opens MRs for them.

I was surprised to find out this is all you need to keep the MRs up-to-date with rebases and all. Gitlab updates the target branch in the existing MR when you try to create the MR with a new target. When you merge a branch, MRs targeting it update accordingly, too.

_As an aside, it's incredible how I can never remember if it's PR or MR on Gitlab._

**Comment by djzort** [comment]
The [gcli](https://github.com/herrhotzenplotz/gcli) may provide a forge neutral intermediate?
